spec/lib/rmt/cli/mirror_spec.rb

```diff
@@ -18,69 +18,45 @@
       end
     end
 
-    context 'repositories marked for mirroring' do
-      let(:repository) { FactoryGirl.create :repository, :with_products, mirroring_enabled: true }
+    context 'with repositories marked for mirroring' do
+      let!(:repository) { create :repository, :with_products, mirroring_enabled: true }
 
       context do
         subject(:command) do
-          repository
           expect_any_instance_of(RMT::Mirror).to receive(:mirror).at_least(:once) do
             puts 'Test double'
           end
-          described_class.mirror
+          described_class.new.repos
           repository.reload
         end
 
         it 'outputs mirroring progress' do
-          expect { command }.to output("Mirroring repository #{repository.name}\nTest double\n").to_stdout.and output('').to_stderr
+          expect { command }.to output(/Mirroring repository #{repository.name}/).to_stdout.and output('').to_stderr
         end
       end
 
       context do
         before do
-          repository
           expect_any_instance_of(RMT::Mirror).to receive(:mirror).at_least(:once)
           allow(STDOUT).to receive(:puts)
-          described_class.mirror
+          described_class.new.repos
           repository.reload
         end
 
         it 'updates repository mirroring timestamp' do
           expect(repository.last_mirrored_at).not_to be_nil
         end
       end
-    end
 
-    context 'with exceptions during mirroring' do
-      subject(:command) do
-        repository
-        expect_any_instance_of(RMT::Mirror).to receive(:mirror).at_least(:once) do
-          raise RMT::Mirror::Exception, 'Test double exception'
+      context 'with exceptions during mirroring' do
+        subject(:command) do
+          expect_any_instance_of(RMT::Mirror).to receive(:mirror).and_raise(RMT::Mirror::Exception, 'black mirror')
```

Comment:
I would move this line to the it block with an allow:
```ruby
allow_any_instance_of(RMT::Mirror).to receive(:mirror).and_raise(RMT::Mirror::Exception, 'black mirror')
```
We don't "expect" RMT::Mirror to raise the exception, what we care about is the output to STDERR.